tonic/src/codec/compression.rs

```diff
@@ -30,22 +38,58 @@ impl EnabledCompressionEncodings {
     }
 
     /// Enable a [`CompressionEncoding`].
+    /// Every time an encoding is enabled, it is given the lowest priority (the start of the list)
+    /// In order to enable both `gzip` and `zstd`, and have zstd have higher priority, you would call:
+    /// `enable(CompressionEncoding::Zstd).enable(CompressionEncoding::Gzip)`
     pub fn enable(&mut self, encoding: CompressionEncoding) {
         match encoding {
             #[cfg(feature = "gzip")]
             CompressionEncoding::Gzip => self.gzip = true,
             #[cfg(feature = "zstd")]
             CompressionEncoding::Zstd => self.zstd = true,
         }
+
+        // If it is already enabled, remove it, so we can add it to the front of the list
+        for encoding_opt in &mut self.order {
+            if let Some(enabled_encoding) = encoding_opt {
+                if *enabled_encoding == encoding {
+                    *encoding_opt = None;
+                }
+            }
+        }
+
+        // Free up the element at the front of the list by shifting all elements to the right
+        for i in (1..self.order.len()).rev() {
```

Comment:
Nit, would something like [rotate_right](https://doc.rust-lang.org/std/primitive.slice.html#method.rotate_right) help here"

Reply:
simplified and also realized I had a bug lol 
```
        // If it is already enabled, shift the other elements to the left (overwriting it)
        if let Some(index) = self.order.iter().position(|&e| e == Some(encoding)) {
            for i in index..self.order.len() - 1 {
                self.order[i] = self.order[i + 1];
            }
        }

        // Free up the element at the front of the list by shifting all elements to the right
        self.order.rotate_right(1);
```